src/sentry/preprod/vcs/status_checks/tasks.py

```diff
@@ -0,0 +1,326 @@
+
+logger = logging.getLogger(__name__)
+
+SIZE_ANALYZER_TITLE = "Sentry Size Analysis"  # TODO(preprod): translate
+
+
+@instrumented_task(
+    name="sentry.preprod.tasks.create_preprod_status_check",
+    queue="integrations",
+    silo_mode=SiloMode.REGION,
+    retry=Retry(times=3),
+    taskworker_config=TaskworkerConfig(
+        namespace=integrations_tasks,
+        processing_deadline_duration=30,
+    ),
+)
+def create_preprod_status_check_task(preprod_artifact_id: int) -> None:
+    try:
+        preprod_artifact = PreprodArtifact.objects.get(id=preprod_artifact_id)
+    except PreprodArtifact.DoesNotExist:
+        logger.exception(
+            "preprod.status_checks.create.artifact_not_found",
+            extra={"artifact_id": preprod_artifact_id},
+        )
+        return
+
+    logger.info(
+        "preprod.status_checks.create.start",
+        extra={"artifact_id": preprod_artifact.id},
+    )
+
+    title = SIZE_ANALYZER_TITLE
+
+    # TODO(preprod): add real formatting
+    if (
+        preprod_artifact.state == PreprodArtifact.ArtifactState.UPLOADING
+        or preprod_artifact.state == PreprodArtifact.ArtifactState.UPLOADED
+    ):
+        status = StatusCheckStatus.IN_PROGRESS
+        text = f"⏳ Build {preprod_artifact.id} for {preprod_artifact.app_id} is processing..."
+        summary = f"Build {preprod_artifact.id} for `{preprod_artifact.app_id}` is processing..."
+        target_url = None
+    elif preprod_artifact.state == PreprodArtifact.ArtifactState.FAILED:
+        status = StatusCheckStatus.FAILURE
+        text = f"❌ Build {preprod_artifact.id} failed. Error: {preprod_artifact.error_message}"
+        summary = f"Build {preprod_artifact.id} for `{preprod_artifact.app_id}` failed."
+        target_url = None
+    elif preprod_artifact.state == PreprodArtifact.ArtifactState.PROCESSED:
+        status = StatusCheckStatus.SUCCESS
+        text = f"✅ Build {preprod_artifact.id} processed successfully."
+        summary = (
+            f"Build {preprod_artifact.id} for `{preprod_artifact.app_id}` processed successfully."
+        )
+        target_url = None
+    else:
+        raise ValueError(f"Invalid artifact state: {preprod_artifact.state}")
+
+    if not preprod_artifact.commit_comparison:
+        logger.info(
+            "preprod.status_checks.create.no_commit_comparison",
+            extra={"artifact_id": preprod_artifact.id},
+        )
+        return
+
+    commit_comparison: CommitComparison = preprod_artifact.commit_comparison
+    if not commit_comparison.head_sha or not commit_comparison.head_repo_name:
+        # if the user provided git information, we should have a head_sha and head_repo_name
+        logger.error(
+            "preprod.status_checks.create.missing_git_info",
+            extra={
+                "artifact_id": preprod_artifact.id,
+                "commit_comparison_id": commit_comparison.id,
+            },
+        )
+        return
```

Comment:
so we'll just silently fail here? wondering if this is an acceptable failure to ignore from the user's POV

Reply:
If we want to show an error message to the user around missing Git fields we'll have to do that when the build is uploaded to us, not in this code. I'm also debating making this just an info log in case someone just wants to send us sha/base_sha for comparisons and skip integrating the Github app.